ListingManager/ListingManager.cs

```diff
@@ -0,0 +1,306 @@
+    {
+        public static ICollection<string> GetAllExtraListings(string pathToStartFrom)
+        {
+            var files = new List<string>();
+            
+            FileManager.GetAllFilesAtPath(pathToStartFrom, true).ToList().ForEach(f =>
+            {
+                if (IsExtraListing(f))
+                {
+                    files.Add(f);
+                }
+            });
+
+            return files;
+        }
+        
+        private static bool IsListing(string listingPath, out ListingInformation listingData)
+        {
+            listingData = null;
+            
+            if (Path.GetExtension(listingPath) != ".cs") return false;
+
+            try
+            {
+                listingData = new ListingInformation(listingPath);
+            }
+            catch (Exception)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static void UpdateChapterListingNumbers(string pathToChapter,
+            bool verboseMode = false, bool preview = false)
+        {
+            var listingData = new List<ListingInformation>();
+            
+            List<string> allListings = FileManager.GetAllFilesAtPath(pathToChapter)
+                .OrderBy(x => x)
+                .Where(x =>
+                {
+                    bool result = IsListing(x, out var data);
+                    if(result) listingData.Add(data);
+                    return result;
+                }).ToList();
+
+            for (int i = 0, listingNumber = 1; i < allListings.Count; i++, listingNumber++)
+            {
+                string cur = allListings[i];
+
+                var curListingData = listingData[i];
+
+                if (listingNumber == curListingData.ListingNumber)
+                {
+                    continue;
+                }
+
+                UpdateListingNamespace(cur, curListingData.ChapterNumber, 
+                    listingNumber+"", 
+                    curListingData.ListingDescription, verboseMode, preview);
+
+                if (GetPathToAccompanyingUnitTest(cur, out string pathToTest))
+                {
+                    Console.Write("Updating test. ");
+                    UpdateListingNamespace(pathToTest, curListingData.ChapterNumber,
+                        listingNumber+"", 
+                        curListingData.ListingDescription, verboseMode, preview);
+                }
+            }
+        }
+        
+        public static bool IsExtraListing(string path,
+            string regexNamespace = @".*Listing\d{2}.\d{2}(A|B|C|D).*.cs$")
+        {
+            Regex fileNameRegex = new Regex(regexNamespace);
+
+            string directoryNameFull = Path.GetDirectoryName(path);
+            int lastIndexOfPathSep = directoryNameFull.LastIndexOf(Path.DirectorySeparatorChar);
+
+            string directoryName = directoryNameFull
+                .Substring(lastIndexOfPathSep, directoryNameFull.Length - lastIndexOfPathSep);
```

Comment:
You can use `Path.GetFileName` to get the name of the last item in a file path string.